plugin/audit_log/audit_log.c

```diff
@@ -697,12 +769,21 @@ void audit_log_notify(MYSQL_THD thd __attribute__((unused)),
       if (event_general->general_command_length == 4 &&
           strncmp(event_general->general_command, "Quit", 4) == 0)
         break;
-      len= audit_log_general_record(buf, sizeof(buf),
+      len= audit_log_general_record(&log_rec, sizeof(buf),
                                     event_general->general_command,
                                     event_general->general_time,
                                     event_general->general_error_code,
                                     event_general);
-      audit_log_write(buf, len);
+      if (len > sizeof(buf))
```

Comment:
Should not be expensive. Basically it `(uchar*)thd->variables.dynamic_variables_ptr + offset` with additional checks.